File: frontend/src/types/release.ts
export interface ReleaseCheckItem {
  name: string
  status: string
  message: string
  blocking: boolean
  detail: Record<string, unknown>
}

export interface ReleaseStatus {
  launcher_available: boolean
  frontend_static_available: boolean
  writable_runtime_dirs: boolean
  offline_readiness: string
  field_profile_saved: boolean
  status: string
  platform: string | null
  python_version: string | null
  app_root: string | null
  writable_logs: boolean
  writable_exports: boolean
  config_loaded: boolean
  model_dir_present: boolean
  active_model_loaded: boolean
  camera_devices_detected: number
  serial_devices_detected: number
  pico_candidate_count: number
  hardware_command_enabled: boolean
  dry_run: boolean
  no_fire: boolean
  safety_invariant_ok: boolean
  release_manifest_path: string | null
  cold_start_evidence: Record<string, unknown>
  suggested_actions: string[]
  checks: ReleaseCheckItem[]
  generated_at: number
  no_physical_command_generated: boolean
}

export interface ReleasePackageRecord {
  package_id: string
  output_dir: string
  zip_path: string
  files_count: number
  checksums_path: string
  manifest_path: string
  commit_hash: string | null
  source_commit: string
  package_generated_commit: string
  package_workflow_commit: string
  report_commit: string
  checksum_status: string
  release_demo_ready: boolean
  competition_ready: boolean
  dataset_ready_for_training: boolean
  no_physical_command_generated: boolean
  safety_invariant: string
  created_at: number
}

export interface CleanroomVerificationRecord {
  run_id: string
  package_id: string
  zip_path: string
  extract_path: string
  launch_command: string
  smoke_status: string
  endpoints_passed: number
  endpoints_total: number
  frontend_dist_present: boolean
  backend_present: boolean
  forbidden_entries: string[]
  secrets_or_tokens: string[]
  launcher_hardcoded_repo_path: boolean
  release_demo_ready: boolean
  competition_ready: boolean
  no_physical_command_generated: boolean
  created_at: number
}

File: backend/app/schemas/release.py
import time

from pydantic import BaseModel, Field


class ReleaseCheckItem(BaseModel):
    name: str
    status: str
    message: str
    blocking: bool = False
    detail: dict = Field(default_factory=dict)


class ReleaseStatus(BaseModel):
    launcher_available: bool
    frontend_static_available: bool
    writable_runtime_dirs: bool
    offline_readiness: str
    field_profile_saved: bool
    status: str = "warning"
    platform: str | None = None
    python_version: str | None = None
    app_root: str | None = None
    writable_logs: bool = False
    writable_exports: bool = False
    config_loaded: bool = False
    model_dir_present: bool = False
    active_model_loaded: bool = False
    camera_devices_detected: int = 0
    serial_devices_detected: int = 0
    pico_candidate_count: int = 0
    hardware_command_enabled: bool = False
    dry_run: bool = True
    no_fire: bool = True
    safety_invariant_ok: bool = True
    release_manifest_path: str | None = None
    cold_start_evidence: dict = Field(default_factory=dict)
    suggested_actions: list[str] = Field(default_factory=list)
    checks: list[ReleaseCheckItem] = Field(default_factory=list)
    generated_at: float = Field(default_factory=time.time)
    no_physical_command_generated: bool = True


class ReleaseManifest(BaseModel):
    commit_hash: str
    phase: str = "Phase 22"
    build_id: str
    generated_at: float = Field(default_factory=time.time)
    platform: str
    included_components: list[str]
    excluded_runtime_dirs: list[str]
    safety_invariant: dict
    launcher_files: list[str]
    frontend_dist_present: bool
    backend_entrypoint: str
    dependency_strategy: str
    no_physical_command_generated: bool = True


class ReleasePackageRecord(BaseModel):
    package_id: str
    output_dir: str
    zip_path: str
    files_count: int
    checksums_path: str
    manifest_path: str
    commit_hash: str | None = None
    source_commit: str
    package_generated_commit: str
    package_workflow_commit: str
    report_commit: str
    checksum_status: str = "passed"
    release_demo_ready: bool = False
    competition_ready: bool = False
    dataset_ready_for_training: bool = False
    no_physical_command_generated: bool = True
    safety_invariant: str = "DISARMED + NO_FIRE + dry_run=true + hardware_enabled=false + physical_command_enabled=false"
    created_at: float = Field(default_factory=time.time)


class CleanroomSmokeEndpoint(BaseModel):
    method: str = "GET"
    path: str
    status_code: int
    ok: bool


class CleanroomVerificationRecord(BaseModel):
    run_id: str
    package_id: str
    zip_path: str
    extract_path: str
    launch_command: str
    smoke_status: str
    endpoints: list[CleanroomSmokeEndpoint] = Field(default_factory=list)
    endpoints_passed: int = 0
    endpoints_total: int = 0
    frontend_dist_present: bool = False
    backend_present: bool = False
    forbidden_entries: list[str] = Field(default_factory=list)
    secrets_or_tokens: list[str] = Field(default_factory=list)
    launcher_hardcoded_repo_path: bool = False
    release_demo_ready: bool = True
    competition_ready: bool = False
    no_physical_command_generated: bool = True
    report_paths: dict[str, str] = Field(default_factory=dict)
    created_at: float = Field(default_factory=time.time)
